packages/flutter/lib/src/painting/continuous_rectangle_border.dart

```diff
@@ -3,58 +3,251 @@
+        ..cubicTo(rightX(0.86840701), bottomY(0),
+            rightX(1.08849299), bottomY(0.0),
+            rightX(1.52866483), bottomY(0.0))
+        ..lineTo(leftX(1.52866483), bottomY(0.0))
+        ..cubicTo(leftX(1.08849323), bottomY(0.0),
+            leftX(0.86840689), bottomY(0.0),
+            leftX(0.66993427), bottomY(0.06549600))
+        ..lineTo(leftX(0.63149399), bottomY(0.07491100))
+        ..cubicTo(leftX(0.37282392), bottomY(0.16905899),
+            leftX(0.16906013), bottomY(0.37282401),
+            leftX(0.07491176), bottomY(0.63149399))
+        ..cubicTo(leftX(0), bottomY(0.86840701),
+            leftX(0.0), bottomY(1.08849299),
+            leftX(0.0), bottomY(1.52866483))
+        ..lineTo(leftX(0.0), topY(1.52866471))
+        ..cubicTo(leftX(0.0), topY(1.08849323),
+            leftX(0.0), topY(0.86840689),
+            leftX(0.06549600), topY(0.66993427))
+        ..lineTo(leftX(0.07491100), topY(0.63149399))
+        ..cubicTo(leftX(0.16905899), topY(0.37282392),
+            leftX(0.37282401), topY(0.16906013),
+            leftX(0.63149399), topY(0.07491176))
+        ..cubicTo(leftX(0.86840701), topY(0),
+            leftX(1.08849299), topY(0.0),
+            leftX(1.52866483), topY(0.0))
+        ..close();
+    }
+
+    // The ratio of the declared corner radius to the total affected pixels
+    // along each axis to render the corner. For example if the declared radius
+    // were 25px then totalAffectedCornerPixelRatio * 25 (~38) pixels would be
+    // affected along each axis.
+    //
+    // It is also the multiplier where the resulting shape will be convex with
+    // a height and width of any value. Below this value, noticeable clipping
+    // will be seen at large rectangle dimensions.
+    //
+    // If the shortest side length to radius ratio drops below this value, the
+    // radius must be lessened to avoid clipping (ie. concavity) of the shape.
+    //
+    // This value comes from the website where the other equations and curves
+    // were found
+    // (https://www.paintcodeapp.com/news/code-for-ios-7-rounded-rectangles).
+    const double totalAffectedCornerPixelRatio = 1.52865;
+
+    // The ratio of the radius to the magnitude of pixels on a given side that
+    // will be used to construct the two corners.
+    const double minimalUnclippedSideToCornerRadiusRatio = 2.0 * totalAffectedCornerPixelRatio;
+
+    // The multiplier of the radius in comparison to the smallest edge length
+    // used to describe the minimum radius for this shape.
+    //
+    // This is multiplier used in the case of an extreme aspect ratio and a
+    // small extent value. It can be less than
+    // 'minimalUnclippedSideToCornerRadiusRatio' because there
+    // are not enough pixels to render the clipping of the shape at this size so
+    // it appears to still be concave (whereas mathematically it's convex).
+    //
+    // This value was determined by an eyeball comparison with the the native
```

Comment:
Then this should be in a cupertino widget rather than be in the painting package.